app/api/contractreader/route.ts

```diff
@@ -0,0 +1,28 @@
+import { NextRequest, NextResponse } from 'next/server';
+import '@/lib/coinbase';
+import { readContract } from '@coinbase/coinbase-sdk';
+
+export async function POST(req: NextRequest) {
+  try {
+    const { contractAddress, abi, method, args, network } = await req.json();
+
+    const result = await readContract({
+      networkId: network,
+      abi: abi,
+      contractAddress: contractAddress as `0x${string}`,
+      method: method,
+      args: args,
+    });
+
+    // If the result is a BigInt, convert it to a string
+    const resultStr = typeof result === 'bigint' ? result.toString() : result;
```

Comment:
i think there may be a problem here if you read a function that returns a nested type, where a nested type is `bigint`.